### PR TITLE
Check if ENV HTTP_X_FORWARDED_HOST is available before using it

### PR DESCRIPTION
**Pull request by melo**

### CODE REVIEW

File: lib/Dancer2/Core/Request.pm [hunk 1]
```diff
@@ -186,7 +186,7 @@ sub BUILDARGS {
 sub host {
     my ($self) = @_;
 
-    if ( $self->is_behind_proxy ) {
+    if ( $self->is_behind_proxy and $self->env->{HTTP_X_FORWARDED_HOST} ) {
```
Comment (ambs) [inline lib/Dancer2/Core/Request.pm]: probably use `exists` instead of autovivificating it?

**Comment by veryrusty** [comment]
Thanks @melo!  Merged with @ambs suggestion plus tests as 765261a.  :dancers: 
( This is _almost identical_ to the suggestion in #915 :smile: )

**Comment by ambs** [comment]
:+1:

**Comment by abeverley** [comment]
The problem with this is that if the header is missing (i.e. the client sent a HTTP/1.0 header without host), then the internal host name is returned in `host`.

I would have thought it would be better returning an empty string in these circumstances, which I presume is the behaviour you would get if you were *not* running behind a proxy and the client sent no host header.

**Comment by abeverley** [comment]
See #935.

**Pull request closed without merging**